accepted/0017-add-funding-support.md

```diff
@@ -0,0 +1,138 @@
+* Provide a means to **view** a package's backers or funding opportunites
+
+##  Rationale & Alternatives
+* Rationale: This is a straightforward and easily implemented solution to take the first step toward supporting Open Source Package Maintainence sustainability
+* Rationale: Building this functionality into the package management layer is the only reliable way to push funding notifications to the vast majority of developers who already use npm
+* Alternative: Continue to rely on third-party, opt-in, tools & platforms that only support a fraction of the desired audience
+* Alternative: Continue debating a larger, more complex schema /w correspondingly more complex tooling
+
+## Implementation
+
+* Add a `funding` field to `package.json`
+  * supports a string URL or object with...
+  * keys that support arbitrary string identifiers
+    * ex. `"sponsor"`, `"sponsors"`, `"donations"`, `"bounties"`, `"contributors"`, `"patrons"` etc.
+  * values that represent a string URL or an array of URLs
+* Add notification at the end of output of package installation that references the number of packages with `funding` defined 
+  * ex. `23 packages are looking for funding. Run "npm fund" to find out more.`
+* Add `--no-fund` flag to opt-out of the funding notification when installing
+* Add `npm fund <pkg>` subcommand: 
+  * if a singular URL is defined for the specified package's `funding` field, it will try to open it using the `--browser` config param (similar to `npm repo <pkg>`)
+  * if an object is defined for the specified package's `funding` field, it will print out the tree of values
+  * if no package is specified, `npm` will try to print out a tree of all the funding references defined in the current project's installed dependencies
+* Add a visual representation for the funding field/value on package pages on `npmjs.com`
+
+
+**Examples of `funding` usage in `package.json`:**
+```
+{
+  ...
+  "funding": "https://www.patreon.com/my-account"
+  ...
+}
+```
+```
+{
+  ...
+  "funding": {
+    "foundations": "https://openjsf.org/"
+  }
+  ...
+}
+```
+```
+{
+  ...
+  "funding": {
+    "corporations": [
+      "https://microsoft.com/",
+      "https://google.com/"
+    ]
+  }
+  ...
+}
+```
+```
+{
+  ...
+  "funding": {
+    "sponsor": [
+      "https://github.com/users/my-account/sponsorship",
+      "https://opencollective.com/my-account",
+      "https://www.patreon.com/my-account"
+    ],
+    "sponsors": "https://github.com/users/my-account/sponsorship#sponsors",
+    "contributors": "https://opencollective.com/my-account#section-contributors",
+    "patrons": [
+      "https://patrons-site-one.com/",
+      "https://patrons-site-two.com/",
+      "https://patrons-site-three.com/"
+    ]
+  }
+  ...
+}
+```
+
+**Example of `npm fund <pkg>`:**
+```
+$ npm fund example-package 
+└─ example-package
+   ├─ sponsor
+   │  ├─ 0: https://github.com/users/my-account/sponsorship
```

Comment:
ideally the indices won’t be displayed here; surely it can be displayed as a list?